src/read/xcoff/symbol.rs

```diff
@@ -70,14 +73,50 @@ where
             (&[][..], StringTable::default())
         };
 
-        Ok(SymbolTable { symbols, strings })
+        Ok(SymbolTable {
+            symbols,
+            strings,
+            header: PhantomData,
+        })
+    }
+
+    /// Return the symbol entry at the given index and offset.
+    pub fn get<T: Pod>(&self, index: usize, offset: usize) -> Result<&'data T> {
+        let entry = index
+            .checked_add(offset)
+            .and_then(|x| self.symbols.get(x))
+            .read_error("Invalid XCOFF symbol index")?;
+        let bytes = bytes_of(entry);
+        Bytes(bytes).read().read_error("Invalid XCOFF symbol data")
     }
 
     /// Return the symbol at the given index.
-    pub fn symbol(&self, index: usize) -> read::Result<&'data Xcoff::Symbol> {
-        self.symbols
-            .get(index)
-            .read_error("Invalid XCOFF symbol index")
+    pub fn symbol(&self, index: usize) -> Result<&'data Xcoff::Symbol> {
+        self.get::<Xcoff::Symbol>(index, 0)
+    }
+
+    /// Return the file auxiliary symbol.
+    pub fn aux_file(&self, index: usize) -> Result<&'data Xcoff::FileAux> {
```

Comment:
`aux_file` is not called from anywhere. Should it be?

Reply:
It might be called if we want to get the source file-name information, source-language ID, CPU-version ID information, etc. And it'll be required for xcoff/readobj. Currently it's not called from anywhere.